supertokens_python/recipe/dashboard/api/userdetails/user_metadata_put.py

```diff
@@ -0,0 +1,68 @@
+import json
+from typing import Any, Dict
+
+from supertokens_python.exceptions import raise_bad_input_exception
+from supertokens_python.recipe.usermetadata import UserMetadataRecipe
+from supertokens_python.recipe.usermetadata.asyncio import (
+    clear_user_metadata,
+    update_user_metadata,
+)
+from supertokens_python.types import APIResponse
+
+from ...interfaces import APIInterface, APIOptions, APIResponse
+
+
+class UserMetadataPutAPIResponse(APIResponse):
+    status: str = "OK"
+
+    def to_json(self) -> Dict[str, Any]:
+        return {"status": self.status}
+
+
+async def handle_metadata_put(
+    _api_interface: APIInterface, api_options: APIOptions
+) -> APIResponse:
+    request_body: Dict[str, Any] = await api_options.request.json()  # type: ignore
+    user_id = request_body.get("userId")
+    data = request_body.get("data")
+
+    # This is to throw an error early in case the recipe has not been initialised
+    UserMetadataRecipe.get_instance()
+
+    if user_id is None or isinstance(user_id, str) is False:
+        raise_bad_input_exception(
+            "Required parameter 'userId' is missing or has an invalid type"
+        )
+
+    if data is None or isinstance(data, str) is False:
+        raise_bad_input_exception(
+            "Required parameter 'data' is missing or has an invalid type"
+        )
+
+    try:
+        parsed_data = json.loads(data)
```

Comment:
Will this just check for valid JSON strings? Or will it check for an object?

Reply:
valid json strings and throws error if it's not parsable.

Reply:
So metadata cannot be arrays, null etc so we need to enforce that it is a JSON object. This is done in the node sdk so you can refer to that

Reply:
Done.